sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java

```diff
@@ -201,8 +201,10 @@ private Field getIdField(Class<?> domainType) {
             throw new IllegalArgumentException("domain should contain @Id field or field named id");
         } else if (idField.getType() != String.class
                 && idField.getType() != Integer.class
-                && idField.getType() != int.class) {
-            throw new IllegalArgumentException("type of id field must be String or Integer");
+                && idField.getType() != int.class
```

Comment:
Can we add a unit test for this in `CosmosEntityInformationUnitTest.java` class ?

Reply:
Added in CosmosEntityInformationUnitTest, please check

Reply:
Thanks.